### PR TITLE
Typos and suggestions

### PR DESCRIPTION
**Pull request by jannes-m**
Dear mlr-team,
congrats on releasing the beta version of mlr3. And it is great to see that you are writing a book on **mlr3**, I will certainly enjoy reading it!
In this PR I have just fixed a few typos and added 2 suggestions. Pls let me know if you want me to report similar stuff from the other chapter, I would be glad to do so.

### CODE REVIEW

File: bookdown/00-intro.Rmd [hunk 1]
```diff
@@ -31,6 +32,7 @@ We follow the general design principles below in the implementation of the `r ml
 * Embrace [R6](https://cran.r-project.org/package=R6), clean object-oriented design, object state-changes, and reference semantics.
 * Embrace [`data.table`](https://cran.r-project.org/package=data.table) for fast and convenient data frame computations.
 * Unify container and result classes as much as possible and provide result data in `data.table`s.
+<!-- maybe replace aggregation by split-apply-combine strategy in accordance with Hadley's paper and as a more general term -->
     This considerably simplifies the API and allows easy selection and aggregation operations.
```
Comment (pat-s) [inline bookdown/00-intro.Rmd]:
```suggestion
    This considerably simplifies the API and allows easy selection and "split-apply-combine" (aggregation) operations.
```

**Comment by pat-s** [comment]
Hey @jannes-m, thanks for looking into it! Contributions are always highly welcome of course! :+1:

File: bookdown/00-intro.Rmd [hunk 2]
```diff
@@ -10,16 +10,17 @@ Examples of this advanced functionality include hyperparameter tuning, feature s
...
 We target both **practitioners** who want to quickly apply machine learning algorithms and **researchers** who want to implement, benchmark, and compare their new methods in a structured environment.
 The package is a complete rewrite of an earlier version of `r mlr_pkg("mlr")` that leverages many years of experience to provide a state-of-the-art system that is easy to use and extend.
-It is intended for users who have basic knowledge in machine learning and R that are interested in complex projects and advanced functionality rather than one-liners for one specific thing.
+It is intended for users who have basic knowledge in machine learning and R and who are interested in complex projects and advanced functionality rather than one-liners for one specific thing.
 
 **Why a Rewrite?**
 
 `r mlr_pkg("mlr")` was first released to CRAN in 2013, with the core design and architecture dating back much further.
 Over time, the addition of many features has led to a considerably more complex design that made it harder to build, maintain, and extend than we had hoped for.
+<!-- data.table v.1.0 was released in 2006, see https://github.com/Rdatatable/data.table/wiki-->
```
Comment (pat-s) [inline bookdown/00-intro.Rmd]: I think back then it was not as popular as it is now :) Thanks for the link!

File: bookdown/00-intro.Rmd [hunk 3]
```diff
@@ -16,11 +16,10 @@ It is intended for users who have basic knowledge in machine learning and R and
 
 `r mlr_pkg("mlr")` was first released to CRAN in 2013, with the core design and architecture dating back much further.
 Over time, the addition of many features has led to a considerably more complex design that made it harder to build, maintain, and extend than we had hoped for.
-<!-- data.table v.1.0 was released in 2006, see https://github.com/Rdatatable/data.table/wiki-->
-Further, the R ecosystem was not as far developed at the time, for example `r cran_pkg("data.table")` did not exist.
-Including such helpful packages makes `r mlr_pkg("mlr3")` easier to maintain, to use, and more performant, but also resulted in non-trivial API changes that would have been hard to implement and support in `r mlr_pkg("mlr")`.
-With hindsight, we saw that some of the design and architecture changes in `r mlr_pkg("mlr")` made it difficult to support new features, in particular with respect to pipelines.
-In 2018, we decided to start working on the reimplementation, and in July 2019 we released the first version of `r mlr_pkg("mlr3")` to CRAN.
+Furthermore, the R ecosystem as well as helpful packages such as `r cran_pkg("data.table")` have undergone major changes in the meantime.
```
Comment (jannes-m) [inline bookdown/00-intro.Rmd]: @pat-s pls have a look if you like my rewrite and if it expresses what you wanted to say

File: bookdown/00-intro.Rmd [hunk 4]
```diff
@@ -10,16 +10,16 @@ Examples of this advanced functionality include hyperparameter tuning, feature s
...
-Including such helpful packages makes `r mlr_pkg("mlr3")` easier to maintain, to use, and more performant, but also resulted in non-trivial API changes that would have been hard to implement and support in `r mlr_pkg("mlr")`.
-With hindsight, we saw that some of the design and architecture changes in `r mlr_pkg("mlr")` made it difficult to support new features, in particular with respect to pipelines.
-In 2018, we decided to start work on the reimplementation, and in July 2019 we released the first version of `r mlr_pkg("mlr3")` to CRAN.
+Furthermore, the R ecosystem as well as helpful packages such as `r cran_pkg("data.table")` have undergone major changes in the meantime.
+It would have been nearly impossible to integrate all these changes into the original design of `r mlr_pkg("mlr")`.
+Instead, we decided to start working on a reimplementation in 2018, which resulted in the first release of `r mlr_pkg("mlr3")` on CRAN .
+The new design and the integration of further developed R packages and features (**data.table**, **R6**, pipelines) makes `r mlr_pkg("mlr3")` much easier to maintain, easier to use, and more performant compared to `r mlr_pkg("mlr")`.
```
Comment (pat-s) [inline bookdown/00-intro.Rmd]:
```suggestion
The new design and the integration of newly developed R packages (**R6**, **future**) makes `r mlr_pkg("mlr3")` much easier to maintain and use, and more performant compared to `r mlr_pkg("mlr")`.
```

File: bookdown/00-intro.Rmd [hunk 5]
```diff
@@ -31,7 +31,8 @@ We follow the general design principles below in the implementation of the `r ml
 * Embrace [R6](https://cran.r-project.org/package=R6), clean object-oriented design, object state-changes, and reference semantics.
 * Embrace [`data.table`](https://cran.r-project.org/package=data.table) for fast and convenient data frame computations.
 * Unify container and result classes as much as possible and provide result data in `data.table`s.
-    This considerably simplifies the API and allows easy selection and aggregation operations.
+<!-- maybe replace aggregation by split-apply-combine strategy in accordance with Hadley's paper and as a more general term -->
```
Comment (pat-s) [inline bookdown/00-intro.Rmd]: This still needs to be removed.

File: bookdown/00-intro.Rmd [hunk 6]
```diff
@@ -16,6 +16,7 @@ It is intended for users who have basic knowledge in machine learning and R and
 
 `r mlr_pkg("mlr")` was first released to CRAN in 2013, with the core design and architecture dating back much further.
 Over time, the addition of many features has led to a considerably more complex design that made it harder to build, maintain, and extend than we had hoped for.
+<!-- data.table v.1.0 was released in 2006, see https://github.com/Rdatatable/data.table/wiki-->
```
[the remark this answers is not in the document]
Comment (pat-s) [reply bookdown/00-intro.Rmd]: Imo it reads fine now with your most recent changes :+1:

File: bookdown/00-intro.Rmd (hunk 4 above)
Comment (jannes-m) [reply bookdown/00-intro.Rmd]: how about `further and newly developed` R packages, this would then also include the **data.table** package.

Comment (pat-s) [reply bookdown/00-intro.Rmd]: Ok. Seems like you need to commit the suggestion, I can't :)

**Comment by jannes-m** [comment]
Ok, I think now it is finally ready to merge. The only thing left to check is if by pipelines you mean the magrittr package. If not, let me know and I delete the reference to the **magrittr** package again. Thanks!

**Comment by berndbischl** [comment]
> Ok, I think now it is finally ready to merge. The only thing left to check is if by pipelines you mean the magrittr package. If not, let me know and I delete the reference to the **magrittr** package again. Thanks

we tried to use magrittr but it didnt work for us. we dont use it currently

**Comment by jannes-m** [comment]
Thanks for clarifying @berndbischl. Consequently, I have deleted the **magrittr** reference. And I will ask @pat-s in person what you mean by pipelines (data.table pipelines, i.e, [][]) next time I see him in person.

**Comment by berndbischl** [comment]
thanks for the help!
but the PR seems to do a bit more than simply correct typos? some sentences are changed pretty much, not always for the better. and if stuff is "unclear" (to the contributor) ask but dont delete?

```
With hindsight, we saw that some of the design and architecture changes in `r mlr_pkg("mlr")` made it difficult to support new features, in particular with respect to pipelines.
```

eg the above as completely removed. and "pipelines" meant "ML pipelines" (the concept) which mlr supported thru its "wrappers" (which was not so great)

**Comment by larskotthoff** [comment]
Thanks for your contribution @jannes-m but it looks to me like at least some of your changes are a matter of opinion rather than correctness and in some cases change the meaning.

I think at this point feedback on things that are not well explained, missing, or incorrect is most welcome. Polishing specific formulations is something we can have a look later when the main content of the book is finished.

**Comment by jannes-m** [comment]
Sorry, I will be more careful. I have re-added the deleted sentence but I can also close the PR if this easier.

**Comment by larskotthoff** [comment]
Great, thanks!

**Pull request merged**